haxepunk/Graphic.hx

```diff
@@ -39,49 +33,35 @@ abstract TileType(Either<BitmapData, TileAtlas>) from Either<BitmapData, TileAtl
  *
  * Conversion is automatic, no need to use this.
  */
-abstract ImageType(Either<BitmapData, AtlasRegion>) from Either<BitmapData, AtlasRegion>
+abstract ImageType(AtlasRegion) from AtlasRegion to AtlasRegion
 {
 	@:dox(hide) @:from public static inline function fromString(s:String):ImageType
 	{
-		if (HXP.renderMode == RenderMode.HARDWARE)
-			return Right(Atlas.loadImageAsRegion(s));
-		else
-			return Left(HXP.getBitmap(s));
+		return Atlas.loadImageAsRegion(s);
 	}
 	@:dox(hide) @:from public static inline function fromTileAtlas(atlas:TileAtlas):ImageType
 	{
-		return Right(atlas.getRegion(0));
+		return atlas.getRegion(0);
 	}
 	@:dox(hide) @:from public static inline function fromAtlasRegion(region:AtlasRegion):ImageType
 	{
-		return Right(region);
+		return region;
 	}
 	@:dox(hide) @:from public static inline function fromBitmapData(bd:BitmapData):ImageType
 	{
-		if (HXP.renderMode == RenderMode.HARDWARE)
-			return Right(Atlas.loadImageAsRegion(bd));
-		else
-			return Left(bd);
+		return Atlas.loadImageAsRegion(bd);
 	}
 
 	public var width(get, never):Int;
 	inline function get_width()
 	{
-		return Std.int(switch (this)
-		{
-			case Left(b): b.width;
-			case Right(a): a.width;
-		});
+		return Std.int(this.width);
```

Comment:
Probably because it uses a Rectangle and width/height are floats in that class. We can cast them in AtlasRegion to int though.